### ISSUE TITLE
Wrong funding fee calc for futures with many position adjustments

### ISSUE THREAD

**Issue by maxkrukov** [freqtrade/freqtrade#9626]
## Describe your environment

  * Operating system: Ubuntu 22.04.3 LTS
  * Python Version: Docker stable-version (2023.12)
  * CCXT version: Docker stable-version (2023.12)
  * Freqtrade Version: Docker stable-version (2023.12)
  
## Describe the problem:

*Wrong funding fee calc for futures with many position adjustments*

### Steps to reproduce:

![image](https://github.com/freqtrade/freqtrade/assets/23353953/8939ac68-2fda-4ecb-9da9-5212b2440feb)
![image](https://github.com/freqtrade/freqtrade/assets/23353953/a5e5fbef-8780-451d-97ff-69aa00ce8d12)
![image](https://github.com/freqtrade/freqtrade/assets/23353953/f4652422-8726-4546-a1c1-5527f7020e60)
![image](https://github.com/freqtrade/freqtrade/assets/23353953/826eecad-2435-45f9-a233-9c30e02daa6f)
Funding fees should be: 3.78985360

**Comment by xmatthias** [comment]
well what makes you believe this? what is `123.csv`? 
what's it's content? 
where does it come from? how did you generate this?

simply catting a file `123.csv` which is not available, where the source is absolutely unclear - as well as the content which is totally unclear doesn't show anything.


Aside from this, what exchange is this trade from?
is it live? dry-run? 
Please also provide all information you think is important as Text (or files), not as screenshots (if i need to start typing your numbers, i'll not look at the issue at all, as i'll waste hours trying type it off correctly).

Also, please provide the output of the following command for the affected trade:
`freqtrade show-trades --db-url sqlite:///<trades.sqlite> --print-json --trade-ids <tradeid>` 
Please use a "details block" to keep the issue itself kindof readable (see below for details on how to do that). 

<details>
<summary>Expand to see how to do details blocks</summary>

```
<details>
<summary>Expand full </summary>
(wrap in tripple ticks `)
Trades output ...
</details>
```
</details>

**Comment by maxkrukov** [comment]
It is live trading on binance.

[123.csv](https://github.com/freqtrade/freqtrade/files/13805259/123.csv)
[Export complete transaction history (2).xlsx](https://github.com/freqtrade/freqtrade/files/13805261/Export.complete.transaction.history.2.xlsx)
Export.complete.transaction.history.2.xlsx is exported history that downloaded from binance UI.
123.csv is converted and shrinked history for sum of funding fees calculation

**Comment by maxkrukov** [comment]
[51-trade.txt](https://github.com/freqtrade/freqtrade/files/13805277/51-trade.txt) - exported trade 51 to json file

**Comment by xmatthias** [comment]
What version was the trade started with?

2023.12 is the current version - but that didn't release until 2 days ago - the trade however is already open since 10+ days...
considering you're running docker, you might still have the previous image available (`docker images`) ... and you should be able to get the version by using `docker run --rm <imageid> -V`

**Comment by xmatthias** [comment]
Wait this is live ... 

Considering the trade is still open - can you please
* enable `log_responses` in the exchange part of the configuration
* Update to the very latest develop (considering you're on docker, please wait about half an hour  - as i added some relevant log-statements for this issue). Please make sure that your new version is either the linked version or a newer version
* restart the bot
* provide me with the logs ... what i'm mainly looking for is `funding_history` - which is a new log-message i added a few seconds ago - but more logs are "usually" good (i can pick the right line myself eventually, and sometimes context is quite helpful :laughing: )

**Comment by maxkrukov** [comment]
> What version was the trade started with?
> 
> 2023.12 is the current version - but that didn't release until 2 days ago - the trade however is already open since 10+ days... considering you're running docker, you might still have the previous image available (`docker images`) ... and you should be able to get the version by using `docker run --rm <imageid> -V`

I ran freqtrade with helm chart in k8s cluster (can provide helm chart). Started from 2023.7 and updated to 2023.12 yesterday. Maybe new trades will be without this issue.

**Comment by maxkrukov** [comment]
> Wait this is live ...
> 
> Considering the trade is still open - can you please
> 
> * enable `log_responses` in the exchange part of the configuration
> * Update to the very latest develop (considering you're on docker, please wait about half an hour  - as i added some relevant log-statements for this issue). Please make sure that your new version is either the linked version or a newer version
> * restart the bot
> * provide me with the logs ... what i'm mainly looking for is `funding_history` - which is a new log-message i added a few seconds ago - but more logs are "usually" good (i can pick the right line myself eventually, and sometimes context is quite helpful 😆 )


```
image:
  base: freqtradeorg/freqtrade       
  tag: "develop"
  imagePullPolicy: Always
....
  exchange:
    name: binance
    log_responses: true
```
[freqtrade.log](https://github.com/freqtrade/freqtrade/files/13806095/freqtrade.log)

**Comment by xmatthias** [comment]
well there's been a _significant_ change in 2023.11 on funding-fee calculations (#9298) - hence the 2023.7 point is surely a point that might cause this.
i was hoping for an update from 2023.11 ... but suspected an earlier version already - as the effect (funding_fees are approximately doubling per order, causing "exploding" funding-fees as order count progresses) is exactly what was observed in that related issue from october.

for "old" orders, it's unfortunately not auto-healing ... so it'll only really affect new orders.

Pretty much what's visible in the last column (this is the funding_fee column on the order object)

![image](https://github.com/freqtrade/freqtrade/assets/5024695/93ca901c-9d61-4078-b063-78598e3d1099)


---

is that "SEI" trade still open? 

Obviously the logic now "works" (at least for this call) - and the funding-fee is now not returning what i hoped for - most likely as there's been an nw order on the 1st (yesterday evening, seen from my timezone).

Would you mind running the below - which i assume will return something rather similar to your "123.csv" file ... 

``` python
import ccxt
exchange = ccxt.binance({
    'apiKey': '<yourApiKey>', 
    'secret': '<yoursecret>', 
    'options': {'defaultType': 'swap'} 
    })
res = exchange.fetch_funding_history(symbol='SEI/USDT:USDT', since=1702951218735)
print(res)
```

**Comment by maxkrukov** [comment]
Yes, it is open. But it seems it`s going for liquidation=)   
[fetch_funding_history.json](https://github.com/freqtrade/freqtrade/files/13811523/fetch_funding_history.json)
```
$ cat fetch_funding_history.json | grep '"income"' | awk -F'"' '{print$4}' | paste -sd+ | bc
4.91880528

```
Thank you for debugging.

**Comment by xmatthias** [comment]
well response is what i expected.

Everything points to this being a duplicate of #9163 - which was fixed in #9298, considering that most of the trade was done with the 2023.7 version (which didn't include this fix yet).

For the moment, i'll assume this as fixed, however i'd ask you to have an eye out for funding fee handling on trades that started with the new version (or at least had all orders starting with the 2nd order with the new version).
If trades with the new version are still showing this issue, i'd like to have a look at data for that trade - for now, there's however nothing i can imagine doing, as it all points to a "already fixed" problem.
